install.sh

```diff
@@ -370,80 +380,155 @@ if [ "$1" = "install" ]; then
-        else
-          _kernelname=$_basekernel.${_kernel_subver}_$_kernel_flavor
-        fi
-        _headers_rpm="kernel-headers-${_kernelname}*.rpm"
-        _kernel_rpm="kernel-${_kernelname}*.rpm"
-        # The headers are actually contained in the kernel-devel RPM and not the headers one...
-        _kernel_devel_rpm="kernel-devel-${_kernelname}*.rpm"
-
-        cd RPMS
-        if [ "$_distro" = "Fedora" ]; then
-          sudo dnf install $_headers_rpm $_kernel_rpm $_kernel_devel_rpm
-        elif [ "$_distro" = "Suse" ]; then
-          msg2 "Some files from 'linux-glibc-devel' will be replaced by files from the custom kernel-hearders package"
-          msg2 "To revert back to the original kernel headers do 'sudo zypper install -f linux-glibc-devel'"
-          sudo zypper install --replacefiles --allow-unsigned-rpm $_headers_rpm $_kernel_rpm $_kernel_devel_rpm
-        fi
-
-        msg2 "Install successful"
+    # Create RPMS folder if it doesn't exist
+    cd "$_where"
+    mkdir -p RPMS
+
+    # Move rpm files to RPMS folder inside the linux-tkg folder
+    mv ${_fedora_work_dir}/RPMS/x86_64/*tkg* "$_where"/RPMS/
+
+    read -p "Do you want to install the new Kernel ? Y/[n]: " _install
+    if [[ "$_install" =~ ^(Y|y|Yes|yes)$ ]]; then
+
+      if [[ "$_sub" = rc* ]]; then
+        _kernelname=$_basekernel.${_kernel_subver}_${_sub}_$_kernel_flavor
+      else
+        _kernelname=$_basekernel.${_kernel_subver}_$_kernel_flavor
       fi
+      _headers_rpm="kernel-headers-${_kernelname}*.rpm"
+      _kernel_rpm="kernel-${_kernelname}*.rpm"
+      # The headers are actually contained in the kernel-devel RPM and not the headers one...
+      _kernel_devel_rpm="kernel-devel-${_kernelname}*.rpm"
+
+      cd RPMS
+      if [ "$_distro" = "Fedora" ]; then
+        sudo dnf install $_headers_rpm $_kernel_rpm $_kernel_devel_rpm
+      elif [ "$_distro" = "Suse" ]; then
+        msg2 "Some files from 'linux-glibc-devel' will be replaced by files from the custom kernel-hearders package"
+        msg2 "To revert back to the original kernel headers do 'sudo zypper install -f linux-glibc-devel'"
+        sudo zypper install --replacefiles --allow-unsigned-rpm $_headers_rpm $_kernel_rpm $_kernel_devel_rpm
+      fi
+
+      msg2 "Install successful"
     fi
 
-  elif [ "$_distro" = "Generic" ]; then
+  elif [[ "$_distro" =~ ^(Gentoo|Generic)$ ]]; then
 
     ./scripts/config --set-str LOCALVERSION "-${_kernel_flavor}"
 
-    if make ${llvm_opt} -j ${_thread_num}; then
+    if [[ "$_sub" = rc* ]]; then
+      _kernelname=$_basekernel.${_kernel_subver}-${_sub}-$_kernel_flavor
+    else
+      _kernelname=$_basekernel.${_kernel_subver}-$_kernel_flavor
+    fi
 
-      if [[ "$_sub" = rc* ]]; then
-        _kernelname=$_basekernel.${_kernel_subver}-${_sub}-$_kernel_flavor
-      else
-        _kernelname=$_basekernel.${_kernel_subver}-$_kernel_flavor
+    msg2 "Building kernel"
+    _timed_build make ${llvm_opt} -j ${_thread_num}
```

Comment:
Can use `make O=./build` here, but you also have to add it here
https://github.com/AdelKS/linux-tkg/blob/a5685a4086b377d4fe70e09b9cf9ed2510ab7340/linux-tkg-config/prepare#L1303
and also fix the location of where the `.config` file would be found.

I have not tested if it's needed in other parts after, hopefully it is enough to find them all.

Reply:
Honestly, it would be so awesome if the kernel had a `cmake` compatibility, where all of those standards are baked in. Also `ninja` builds for faster compilation.

Reply:
Well, the kernel is super old and developed its own [build system](https://www.kernel.org/doc/html/latest/kbuild/index.html). Seeing that change is very unlikely.